hpack/hpack.py

```diff
@@ -241,11 +243,15 @@ def _encode_indexed_literal(self, index, value, indexbit, huffman=False):
 
     def _encode_table_size_change(self):
         """
-        Produces the encoded form of a header table size change context update.
+        Produces the encoded form of all header table size change context updates
         """
-        size_bytes = encode_integer(self.header_table_size, 5)
-        size_bytes[0] |= 0x20
-        return bytes(size_bytes)
+        block = b''
+        while(len(self.table_size_changes) > 0):
+            size_bytes = self.table_size_changes.popleft()
+            size_bytes = encode_integer(size_bytes, 5)
+            size_bytes[0] |= 0x20
+            block += bytes(size_bytes)
```

Comment:
So this `while` loop is not idiomatic. Rather than repeatedly query the `len` and popleft, we can just iterate over the `table_size_changes` container directly and then resize at the end:

```python
for change in self.table_size_changes:
    do_stuff()
self.table_size_changes = collections.deque()
```

Once we make that change we're no longer popping from the left so the advantage of `collections.deque` is gone, so we should also just replace that with a list (cheaper in memory).